tempesta_fw/sock_srv.c

```diff
@@ -414,6 +414,8 @@ tfw_sock_srv_disconnect(TfwConn *conn)
 	if (atomic_read(&conn->refcnt) != TFW_CONN_DEATHCNT)
 		ret = ss_close_sync(sk, true);
 
+	tfw_connection_release(conn);
```

Comment:
I believe that the fix is correct. However, an explanation of the fix is completely lost. The code interactions are very complex, so an explanation needs to be put somewhere. Perhaps as a comment to the `tfw_sock_srv_disconnect()` function (best!), and possibly as a comment to the issue itself on GitHub and in the commit message.

I think that putting a crash trace in the commit message is overkill.

My understanding of the issue is as follows:
* There are two different ways of closing a server connection. One is closing by a backend. The other is closing by Tempesta. These two different ways of a close go through slightly different code paths.
    * A close by a backend is considered to be temporary. In that case all pending requests in the queue of a connection are resent to the backend if the connection is restored relatively quickly, or rescheduled to other connections. In other words, the requests are not lost, and the communication between clients and backends is not stopped.
    * A close by Tempesta on shutdown is considered permanent. All resources are released, which includes all pending requests.
* Closing of a connection is a competitive process as it can be invoked concurrently by either a backend or Tempesta. The code is deliberately written in such a way so that only one of concurrent closes is allowed to proceed. Given that, there's a chance that even when Tempesta is in shutdown a request for a close from a backend gets through a tad bit earlier, so the process of closing would go on an undesired path. The request for a close from Tempesta won't get through. Therefore there's the need to assure that the connection's resources are definitely released in this case. Hence the fix.